src/Control/Carrier/Class.hs

```diff
@@ -1,11 +1,40 @@
-{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE FlexibleInstances, FunctionalDependencies #-}
 module Control.Carrier.Class
 ( Carrier(..)
 ) where
 
 import Control.Effect.Class
+import Control.Effect.Choose (Choose(..))
+import Control.Effect.Empty (Empty(..))
+import Control.Effect.Error (Error(..))
+import Control.Effect.NonDet (NonDet)
+import Control.Effect.Reader (Reader(..))
+import Control.Effect.Sum ((:+:)(..))
+import Control.Effect.Writer (Writer(..))
+import Control.Monad ((<=<))
 
 -- | The class of carriers (results) for algebras (effect handlers) over signatures (effects), whose actions are given by the 'eff' method.
 class (HFunctor sig, Monad m) => Carrier sig m | m -> sig where
   -- | Construct a value in the carrier for an effect signature (typically a sum of a handled effect and any remaining effects).
   eff :: sig m a -> m a
+
+
+instance Carrier Empty Maybe where
+  eff Empty = Nothing
+
+instance Carrier (Error e) (Either e) where
+  eff (Throw e)     = Left e
+  eff (Catch m h k) = either (k <=< h) k m
```

Comment:
I called this branch `base-instances` both because they’re instances for types exported from `base`—`Prelude`, even!—and because, like `PureC` and `LiftC`, they’re usable as the _base_ of an effectful context. The signature does not mention `:+:`, since they’re not monad transformers.

Reply:
Ah, so that means I could have an effect stack `State s :+: Error e` and discharge it into an `Either s a` with just a `runState`?

Reply:
Yep, exactly.